include/pybind11/numpy.h

```diff
@@ -244,7 +244,8 @@ template <typename T> struct is_complex : std::false_type { };
 template <typename T> struct is_complex<std::complex<T>> : std::true_type { };
 
 template <typename T> using is_pod_struct = all_of<
-    std::is_pod<T>, // since we're accessing directly in memory we need a POD type
+    std::is_standard_layout<T>,     // since we're accessing directly in memory we need a standard layout type
+    std::is_trivially_copyable<T>,
```

Comment:
For libstcd++ < 5 (which doesn't have `is_trivially_copyable`) this workaround might work:
```C++
    std::is_standard_layout<T>,
#if !defined(__GNUG__) || defined(__clang__) || __GNUC__ >= 5
    std::is_trivially_copyable<T>,
#else
    std::is_trivially_destructible<T>, std::has_trivial_copy_constructor<T>,
#endif
    satisfies_none_of<...
```
It's not *quite* the same as trivially copyable, but it ought to be close enough in practice.